### PR TITLE
DEV2-3020: settings handler

### PR DESCRIPTION
**Pull request by yairco1990**

### CODE REVIEW

File: Common/src/main/java/com/tabnineCommon/chat/actions/toolWindowActions/ChatSettingsAction.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.tabnineCommon.chat.actions.TabnineActionRequest
+import com.tabnineCommon.chat.actions.TabnineChatAction
+import com.tabnineCommon.chat.actions.toolWindowActions.MoveToViewPayload
+
+class ChatSettingsAction(browser: ChatBrowser) :
+    TabnineChatAction(
+        browser,
+        "Settings",
+        "Settings",
+        IconManager.getInstance().getIcon("/icons/settings.svg", ChatSettingsAction::class.java)
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/chat/actions/toolWindowActions/ChatSettingsAction.kt:15]: use `AllIcons.Ide.Notification.Gear`
image

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/ChatSettings.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.tabnineCommon.chat.commandHandlers.chatSettings
+
+import com.google.gson.Gson
+import com.intellij.ide.util.PropertiesComponent
+
+private const val CHAT_SETTINGS_KEY = "com.tabnine.chat.settings"
+
+data class ChatSettingsProps(val isTelemetryEnabled: Boolean?)
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/ChatSettings.kt:8]:
```suggestion
data class ChatSettingsProps(val isTelemetryEnabled: Boolean? = null)
```

this way you don't have to pass `null` to the constructor (`ChatSettingsProps(null)` -> `ChatSettingsProps()`)

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/UpdateChatSettingsHandler.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+import com.intellij.openapi.project.Project
+import com.tabnineCommon.chat.commandHandlers.ChatMessageHandler
+
+class UpdateChatSettingsHandler(gson: Gson) : ChatMessageHandler<ChatSettingsProps, Unit>(gson) {
+    private val chatSettings = ChatSettings(gson)
+
+    override fun handle(payload: ChatSettingsProps?, project: Project) {
+        payload?.let { chatSettings.save(it) }
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/UpdateChatSettingsHandler.kt:12]: warn-log or something if the payload is not present - it should be present, and if not the state will not be set properly and we won't know why

**Review by yonip23 — APPROVED** [review]
lgtm, see comments

File: Common/src/main/java/com/tabnineCommon/chat/actions/toolWindowActions/ChatSettingsAction.kt (hunk 1 above)
Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/chat/actions/toolWindowActions/ChatSettingsAction.kt:15]: There is no gear in older versions...

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/UpdateChatSettingsHandler.kt (hunk 3 above)
Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/chat/commandHandlers/chatSettings/UpdateChatSettingsHandler.kt:12]: I don't see we log this kind of warning on the other handlers. is this necessary? bc with the correct types, the payload should exist in this command.

**Review by dimacodota — APPROVED** [review]

**Pull request merged**